OpenRA.Game/Map/MapPlayers.cs

```diff
@@ -0,0 +1,73 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA
+{
+	public class MapPlayers
+	{
+		public Dictionary<string, PlayerReference> Players;
+
+		public MapPlayers() : this(new List<MiniYamlNode>()) { }
+
+		public MapPlayers(IEnumerable<MiniYamlNode> playerDefinitions)
+		{
+			Players = playerDefinitions.Select(pr => new PlayerReference(new MiniYaml(pr.Key, pr.Value.Nodes)))
+				.ToDictionary(player => player.Name);
+		}
+
+		public MapPlayers(Ruleset rules, int playerCount)
+		{
+			var firstRace = rules.Actors["world"].Traits
+				.WithInterface<CountryInfo>().First(c => c.Selectable).Race;
+
+			if (!Players.ContainsKey("Neutral"))
```

Comment:
`Players` doesn't appear to have been initialized yet, so this will probably NRE?